core/analysis/multi_delimited_token_stream.hpp

```diff
@@ -0,0 +1,67 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "analyzers.hpp"
+#include "token_attributes.hpp"
+#include "utils/attribute_helper.hpp"
+
+namespace irs {
+namespace analysis {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief an analyzer capable of breaking up delimited text into tokens
+///        separated by a set of strings.
+////////////////////////////////////////////////////////////////////////////////
+class multi_delimited_token_stream
+  : public TypedAnalyzer<multi_delimited_token_stream>,
+    private util::noncopyable {
+ public:
+  struct options {
+    std::vector<bstring> delimiters;
+  };
+
+  static constexpr std::string_view type_name() noexcept {
+    return "multi-delimiter";
+  }
+  static void init();
+
+  static analyzer::ptr make(options&&);
+
+  attribute* get_mutable(irs::type_info::type_id type) noexcept final {
+    return irs::get_mutable(attrs_, type);
+  }
+
+  bool reset(std::string_view data) final {
+    data_ = ViewCast<byte_type>(data);
+    return true;
+  }
+
+ protected:
+  using attributes = std::tuple<increment,
+                                offset,  // token value with evaluated quotes
```

Comment:
Please supports offset, offset is begin and end indexes of `term_attribute->value` in the original `data`:

"a_ce", delimiter "_"
"a" -- 0, 1
"ce" -- 2, 4